pkg/kubelet/kubelet.go

```diff
@@ -1942,7 +1943,18 @@ func (kl *Kubelet) canAdmitPod(pods []*v1.Pod, pod *v1.Pod) (bool, string, strin
 	// TODO: out of resource eviction should have a pod admitter call-out
 	attrs := &lifecycle.PodAdmitAttributes{Pod: pod, OtherPods: pods}
 	for _, podAdmitHandler := range kl.admitHandlers {
+		if klog.V(4).Enabled() {
+			klog.V(4).InfoS("Admit pod",
+				"admitHandler", reflect.TypeOf(podAdmitHandler),
+				"pod", klog.KObj(pod),
+				"podUID", pod.UID)
+		}
 		if result := podAdmitHandler.Admit(attrs); !result.Admit {
+			klog.InfoS("Pod is rejected by AdmitHandler",
+				"admitHandler", reflect.TypeOf(podAdmitHandler),
+				"pod", klog.KObj(pod),
+				"podUID", pod.UID,
+				"result", result)
```

Comment:
We should probably reconsider the use of log level 4 (debug log level) here and probably use log level 2 or log by default as these are useful log messages. Also, rather than logging for all admit handlers, I would recommend that you log once before returning from the function and follow the logging convention of logging messages in a single line:
```suggestion
	for _, podAdmitHandler := range kl.admitHandlers {
		if result := podAdmitHandler.Admit(attrs); !result.Admit {
		        klog.InfoS("Pod is rejected by AdmitHandler", "admitHandler", reflect.TypeOf(podAdmitHandler),"pod", klog.KObj(pod), "podUID", pod.UID,  "result", result)
			return false, result.Reason, result.Message
		}
	}
	
	kklog.InfoS("Pod Admitted", "pod", klog.KObj(pod), "podUID", pod.UID)
	return true, "", ""
```

Reply:
Thanks for your suggestion!

Reply:
@swatisehgal `kklog.InfoS("Pod Admitted", "pod", klog.KObj(pod), "podUID", pod.UID)` only prints the admitted pod, but have no handler name, it's also useful if we know the pod has been admitted by which admit handler.